actionpack/test/dispatch/response_test.rb

```diff
@@ -396,9 +396,9 @@ def test_only_set_charset_still_defaults_to_text_html
   test "[response.to_a].flatten does not recurse infinitely" do
     Timeout.timeout(1) do # use a timeout to prevent it stalling indefinitely
       status, headers, body = [@response.to_a].flatten
```

Comment:
This issue was actually resolved in Rack 3 and is no longer possible, so in theory we could just remove this entire test.

Reply:
https://github.com/rack/rack/pull/1453#issuecomment-571440325 for reference.